ui/src/app/views/dataset-page.tsx

```diff
@@ -560,11 +544,11 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
                 based on the variable and value you selected above
               </div>
               </div>
-              <Button data-test-id='save-button' style={{marginRight: '1rem'}}
-                      onClick={this.editing ? () => this.updateDataSet() :
-                () => this.setState({openSaveModal: true})}
-                disabled={this.disableSave() || (this.editing && !dataSetTouched)}>
-                {this.editing ? 'UPDATE DATA SET' : 'SAVE DATA SET'}
+              <Button data-test-id='save-button' style={{position: 'absolute', right: '1rem',
+                top: '.25rem'}} onClick ={() => this.setState({openSaveModal: true})}
+                disabled={this.disableSave()}>
+                {this.editing ? !dataSetTouched ? 'Analyze' :
+                    'Update And Analyze' : 'Save And Analyze'}
```

Comment:
Clicked around here to test and it looks good.  One minor thing I noticed that's probably outside of the scope of this ticket:
when the user goes into an existing data set, the button shows `Analyze`. Then if the user makes a change, the button changes to `Update And Analyze`. This is great. If the user undoes the change they made, then there is no change to the data set but the button still says `Update And Analyze`. As a user, I would expect it to go back to saying `Analyze`, but this might be much more involved than what we want to do right now.

Reply:
Good point. Thats happening because we are just changing on the basis of any action rather than what the action actual is (like select or unselect rather than if the data set options is same as original). 
we/team can discuss more on this, and if we agree it should be another story as the edit indicator is same as what we currently have on test (update button is enabled even after selecting and unselecting)

Reply:
I don't think it would be bad for us to change this here. It shouldn't be hard to store an immutable dataSet in state, and then do a deep compare to see if its been touched. (And you can avoid changing any other code by making `get dataSetTouched` return the deep compare)